src/makeActionCreator.spec.lua

```diff
@@ -0,0 +1,69 @@
+return function()
+	local makeActionCreator = require(script.Parent.makeActionCreator)
+
+	it("should set the name of the actionCreator creator", function()
+		local FooAction = makeActionCreator("foo", function()
+			return {}
+		end)
+
+		expect(FooAction.name).to.equal("foo")
+	end)
+
+	it("should return a table when called as a function", function()
+		local FooAction = makeActionCreator("foo", function()
+			return {}
+		end)
+
+		expect(FooAction()).to.be.a("table")
+	end)
+
+	it("should set the type of the action creator", function()
+		local FooAction = makeActionCreator("foo", function()
+			return {}
+		end)
+
+		expect(FooAction().type).to.equal("foo")
+	end)
+
+	it("should set values", function()
+		local FooAction = makeActionCreator("foo", function(value)
+			return {
+				value = value
+			}
+		end)
+
+		expect(FooAction(100).value).to.equal(100)
+	end)
+
+	it("should throw when its result does not return a table", function()
+		local FooAction = makeActionCreator("foo", function()
+			return function() end
+		end)
+
+		expect(FooAction).to.throw()

```

Comment:
```suggestion
		expect(FooAction).to.throw("must return a table")

```
nvm this is using a 3yr old version of test ez